src/handlers/tsonAsyncIterable.ts

```diff
@@ -23,26 +23,24 @@ export const tsonAsyncIterator: TsonAsyncType<
 	async: true,
 	deserialize: (opts) => {
 		return (async function* generator() {
-			try {
-				for await (const value of opts.stream) {
-					switch (value[0]) {
-						case ITERATOR_DONE: {
-							return;
-						}
+			let next: ReadableStreamReadResult<SerializedIteratorResult>;
+			while (((next = await opts.reader.read()), !next.done)) {
+				switch (next.value[0]) {
+					case ITERATOR_DONE: {
+						opts.controller.close();
+						return;
+					}
 
-						case ITERATOR_ERROR: {
-							throw TsonPromiseRejectionError.from(value[1]);
-						}
+					case ITERATOR_ERROR: {
+						opts.controller.close();
+						throw TsonPromiseRejectionError.from(next.value[1]);
+					}
```

Comment:
I guess we need to test this with that warning

Reply:
Yeah we need to add tests for
- tsonAsyncIterable
- more *weird cases* for deserializeAsync to make sure everything errors and closes correctly